### ISSUE TITLE
Lektor language, .lr extension (as markdown) and .lektorproject (as ini) not recognized by linguist

### ISSUE THREAD

**Issue by CAM-Gerlach** [github/linguist#4151]
## Preliminary Steps

Please confirm you have...
- [x] reviewed [How Linguist Works](https://github.com/github/linguist#how-linguist-works),
- [x] reviewed the [Troubleshooting](https://github.com/github/linguist#troubleshooting) docs,
- [x] considered implementing an [override](https://github.com/github/linguist#overrides),
- [x] verified an issue has not already been logged for your issue ([linguist issues](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+repo%3Agithub/linguist)).

## Problem Description

I'm on the [Spyder IDE](https://github.com/spyder-ide) dev team, and I noticed Lektor ``.lr`` files are not being treated as prose as they should be (they're basically markdown, with extra tags/sections). Furthermore, ``.lektorproject`` files should be treated as ``.ini`` format. [Lektor](https://www.getlektor.com/) is an increasingly popular open-source static site generator, with development hosted on [Github](https://github.com/lektor/lektor), that is often used to deploy sites from/to Github repos, as they do with their own, and used with your Github pages, as [we do](https://github.com/spyder-ide/website-spyder) for the new [Spyder website](https://www.spyder-ide.org/). 

Lack of support is a major problem right now, since aside from manual workarounds it is very difficult to review PRs that don't have manual hard breaks applied, since Github treats the diffs as code and doesn't soft wrap them (unlike even ``git diff`` on the command line). There is a PR to add this on this repo, #2784 , but it was closed due to lack of popularity at the time (lektor had just been first released).

Lektor's popularity has grown tremendously since this PR was initially closed, as it is now used on [441 unique repositories on Github](https://github.com/search?p=1&q=extension%3Alektorproject+NOT+imsurenofilecontainsthis&ref=searchresults&type=Code&utf8=%E2%9C%93) (judging by each needing to have one, and only one ``.lektorproject`` file) and [a total of 30,000 .lr files](https://github.com/search?q=extension%3Alr+NOT+imsurenofilecontainsthisfdsdjds&type=Code), and that number is continuing to grow especially as lektor integrates well with Github Pages. Also, a search of this repo revealed no other/conflicting use of lektor, ``.lr``, or ``.lektorproject``.

Therefore, can the PR be re-opened and merged to add lektor support? I can contribute samples (under MIT) if needed, or re-open a new PR doing the same thing but the author said they'd come back to it when it was more popular, so better to give them a chance first. Let me know what else I can do. Thanks!

**Comment by lildude** [comment]
Popularity looks good now, however that old PR is very out of date.

@mitsuhiko if you're happy to update your PR to reflect all the new requirements for adding a language, and have the bandwidth to do it, please let me know and I'll reopen your PR.

If you can't, don't want to or whatever, let us know and someone else (@CAM-Gerlach you're welcome to it 😉) can open a new PR.

**Comment by CAM-Gerlach** [comment]
Thanks! 

What I would suggest, and would implement if @mitsuhiko isn't able to update their PR, would be to simply add ``.lr`` as an extension for Markdown and add a suitable sample file (and the same for ``.lektorproject`` as @mitsuhiko has done). That would be less extra overhead on both the developer time and the linguist side, and would have the same practical benefits, except that ``Lektor`` wouldn't show up separately in the search results or the language breakdown (which is tangential to the use case presented here, i.e. a huge usability boost using Github in actually developing/reviewing it). If the former was later desired, then it could always be added if someone had the bandwidth.

**Comment by CAM-Gerlach** [comment]
@pchaigno Going ahead now. Did you close the issue by accident?

**Comment by pchaigno** [comment]
> Did you close the issue by accident?

No. We already have more issues than we can handle; we unfortunately can't track all language support suggestions.

> going ahead now.

Great! If you find time to submit the PR before end of day, I'll review right ahead :-)

**Comment by CAM-Gerlach** [comment]
> Great! If you find time to submit the PR before end of day, I'll review right ahead :-)

Done, as #4221

> No. We already have more issues than we can handle; we unfortunately can't track all language support suggestions.

I just found it quite odd that you would label the issue as a good first issue, write a comment explaining such and identify a user who said they would take care of it (and offer it to others if not), and then immediately close it without waiting for anyone to respond. Typically, for an issue like this that is too niche for us to add/fix ourselves, but a community member steps up and says they'll take responsibility for it and get it done in a reasonable time, then we leave it open for them and they close it with their PR. But then again, maybe that has something to do with why we have 10x the open issues as this project with only 2x the total number overall... :)

**Comment by CAM-Gerlach** [comment]
I see; thanks for the explanation! Very true...
